digitalocean/datasource_digitalocean_database_replica_test.go

```diff
@@ -0,0 +1,96 @@
+package digitalocean
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"github.com/digitalocean/godo"
+
+	"github.com/hashicorp/terraform-plugin-sdk/v2/helper/resource"
+	"github.com/hashicorp/terraform-plugin-sdk/v2/terraform"
+)
+
+func TestAccDataSourceDigitalOceanDatabaseReplica_Basic(t *testing.T) {
+	var databaseReplica godo.DatabaseReplica
+	var database godo.Database
+
+	databaseName := randomTestName()
+	databaseReplicaName := randomTestName()
+
+	databaseConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseClusterConfigBasic, databaseName)
+	replicaConfig := fmt.Sprintf(testAccCheckDigitalOceanDatabaseReplicaConfigBasic, databaseReplicaName)
+	datasourceReplicaConfig := fmt.Sprintf(testAccCheckDigitalOceanDatasourceDatabaseReplicaConfigBasic, databaseReplicaName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { testAccPreCheck(t) },
+		ProviderFactories: testAccProviderFactories,
+		CheckDestroy:      testAccCheckDigitalOceanDatabaseReplicaDestroy,
+		Steps: []resource.TestStep{
+			{
+				Config: databaseConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanDatabaseClusterExists("digitalocean_database_cluster.foobar", &database),
+					resource.ComposeTestCheckFunc(
+						func(s *terraform.State) error {
+							time.Sleep(30 * time.Second)
+							return nil
+						},
+					),
+				),
+			},
+			{
+				Config: databaseConfig + replicaConfig,
+				Check: resource.ComposeTestCheckFunc(
+					testAccCheckDigitalOceanDatabaseReplicaExists("digitalocean_database_replica.read-01", &databaseReplica),
+					testAccCheckDigitalOceanDatabaseReplicaAttributes(&databaseReplica, databaseReplicaName),
+					resource.ComposeTestCheckFunc(
+						func(s *terraform.State) error {
+							time.Sleep(30 * time.Second)
```

Comment:
Is the sleep needed here? Would have thought that once the replica is successfully created, the data source should be able to reach it without the sleep.

Reply:
Oh I had that in for testing and was meant to remove it. good catch!